internal/proxy/multi_rate_limiter.go

```diff
@@ -157,36 +159,52 @@ func (rl *rateLimiter) printRates(rates []*internalpb.Rate) {
 
 // registerLimiters register limiter for all rate types.
 func (rl *rateLimiter) registerLimiters() {
+	quotaConfig := &Params.QuotaConfig
 	for rt := range internalpb.RateType_name {
-		var r float64
+		var r *paramtable.ParamItem
 		switch internalpb.RateType(rt) {
 		case internalpb.RateType_DDLCollection:
-			r = Params.QuotaConfig.DDLCollectionRate.GetAsFloat()
+			r = &quotaConfig.DDLCollectionRate
 		case internalpb.RateType_DDLPartition:
-			r = Params.QuotaConfig.DDLPartitionRate.GetAsFloat()
+			r = &quotaConfig.DDLPartitionRate
 		case internalpb.RateType_DDLIndex:
-			r = Params.QuotaConfig.MaxIndexRate.GetAsFloat()
+			r = &quotaConfig.MaxIndexRate
 		case internalpb.RateType_DDLFlush:
-			r = Params.QuotaConfig.MaxFlushRate.GetAsFloat()
+			r = &quotaConfig.MaxFlushRate
 		case internalpb.RateType_DDLCompaction:
-			r = Params.QuotaConfig.MaxCompactionRate.GetAsFloat()
+			r = &quotaConfig.MaxCompactionRate
 		case internalpb.RateType_DMLInsert:
-			r = Params.QuotaConfig.DMLMaxInsertRate.GetAsFloat()
+			r = &quotaConfig.DMLMaxInsertRate
 		case internalpb.RateType_DMLDelete:
-			r = Params.QuotaConfig.DMLMaxDeleteRate.GetAsFloat()
+			r = &quotaConfig.DMLMaxDeleteRate
 		case internalpb.RateType_DMLBulkLoad:
-			r = Params.QuotaConfig.DMLMaxBulkLoadRate.GetAsFloat()
+			r = &quotaConfig.DMLMaxBulkLoadRate
 		case internalpb.RateType_DQLSearch:
-			r = Params.QuotaConfig.DQLMaxSearchRate.GetAsFloat()
+			r = &quotaConfig.DQLMaxSearchRate
 		case internalpb.RateType_DQLQuery:
-			r = Params.QuotaConfig.DQLMaxQueryRate.GetAsFloat()
+			r = &quotaConfig.DQLMaxQueryRate
 		}
-		limit := ratelimitutil.Limit(r)
-		burst := r // use rate as burst, because Limiter is with punishment mechanism, burst is insignificant.
+		limit := ratelimitutil.Limit(r.GetAsFloat())
+		burst := r.GetAsFloat() // use rate as burst, because Limiter is with punishment mechanism, burst is insignificant.
 		rl.limiters[internalpb.RateType(rt)] = ratelimitutil.NewLimiter(limit, burst)
+		onEvent := func(rateType internalpb.RateType) func(*config.Event) {
+			return func(event *config.Event) {
+				f, err := strconv.ParseFloat(event.Value, 64)
+				if err != nil {
+					log.Info("Error format for rateLimit",
+						zap.String("rateType", rateType.String()),
+						zap.String("key", event.Key),
+						zap.String("value", event.Value),
+						zap.Error(err))
+					return
+				}
+				rl.limiters[rateType].SetLimit(ratelimitutil.Limit(f))
+			}
```

Comment:
Map access is not protected by mutex. Maybe using `typeutil.ConcurrentMap` is a better choice?